Failing unit tests in master

I am getting a few unit test failures on current master (on Linux):

```
(nengodev)CTN02:nengo|master> py.test --pyargs nengo                                                                                               4345d1081
==================================================================== test session starts ====================================================================
platform linux2 -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 556 items 

nengo/networks/tests/test_actionselection.py ..
nengo/networks/tests/test_circularconv.py .......
nengo/networks/tests/test_ensemblearray.py ........
nengo/networks/tests/test_integrator.py .
nengo/networks/tests/test_oscillator.py .
nengo/networks/tests/test_product.py .
nengo/networks/tests/test_workingmemory.py F
nengo/spa/tests/test_action_objects.py ........
nengo/spa/tests/test_actions.py ......
nengo/spa/tests/test_assoc_mem.py .....
nengo/spa/tests/test_basalganglia.py ..
nengo/spa/tests/test_buffer.py ....
nengo/spa/tests/test_compare.py ..
nengo/spa/tests/test_cortical.py ......
nengo/spa/tests/test_input.py ...
nengo/spa/tests/test_memory.py .....
nengo/spa/tests/test_pointer.py ...........F.....
nengo/spa/tests/test_thalamus.py s..
nengo/spa/tests/test_vocabulary.py .........
nengo/tests/test_base.py ..
nengo/tests/test_builder.py ........
nengo/tests/test_cache.py ........................s
nengo/tests/test_config.py ........
nengo/tests/test_connection.py .............F.....................................................
nengo/tests/test_dists.py ..................
nengo/tests/test_ensemble.py ............................................
nengo/tests/test_examples.py .....ss..s.s..s..s...s..ss.....s.s.ss........ss..s.s..s..s...s..ss.....s.s.ss...
nengo/tests/test_learning_rules.py ....................
nengo/tests/test_network.py ....
nengo/tests/test_neurons.py ..............
nengo/tests/test_node.py .............
nengo/tests/test_params.py ...............
nengo/tests/test_probe.py .........
nengo/tests/test_simulator.py ........
nengo/tests/test_solvers.py ..........s...sssssssssssssssssss
nengo/tests/test_synapses.py ...s.
nengo/utils/tests/test_builder.py .
nengo/utils/tests/test_builder_graphviz.py .
nengo/utils/tests/test_builder_passthrough.py ...
nengo/utils/tests/test_cache.py ...
nengo/utils/tests/test_compat.py .
nengo/utils/tests/test_connection.py ........
nengo/utils/tests/test_ensemble.py .................
nengo/utils/tests/test_filter_design.py ss
nengo/utils/tests/test_functions_piecewise.py .......
nengo/utils/tests/test_functions_whitenoise.py ..
nengo/utils/tests/test_graphs.py ...
nengo/utils/tests/test_logging.py ..
nengo/utils/tests/test_magic.py ......
nengo/utils/tests/test_nco.py .......
nengo/utils/tests/test_network.py .
nengo/utils/tests/test_neurons.py s.s
nengo/utils/tests/test_numpy.py ....
nengo/utils/tests/test_probe.py ....
nengo/utils/tests/test_progress.py ...........
nengo/utils/tests/test_stdlib.py ......

========================================================================= FAILURES ==========================================================================
___________________________________________________________________ test_inputgatedmemory ___________________________________________________________________

Simulator = <class 'nengo.simulator.Simulator'>, plt = <nengo.utils.testing.Mock object at 0x3ce4cd0>, seed = 2143026370

    def test_inputgatedmemory(Simulator, plt, seed):
        to_memorize = 0.5
        start_memorizing = 0.4
        with nengo.Network(seed=seed) as net:
            test_input = nengo.Node(piecewise(
                {0.0: 0, 0.1: to_memorize, start_memorizing + 0.1: 0}))
            gate_input = nengo.Node(piecewise({0.0: 0, start_memorizing: 1}))
            reset_input = nengo.Node(0)

            mem = nengo.networks.InputGatedMemory(100, 1, difference_gain=5.0)
            nengo.Connection(test_input, mem.input)
            nengo.Connection(gate_input, mem.gate)
            nengo.Connection(reset_input, mem.reset)

            mem_p = nengo.Probe(mem.output, synapse=0.01)

        sim = Simulator(net)
        sim.run(0.5)

        data = sim.data[mem_p]
        t = sim.trange()

        plt.title("gating at %.1f s" % start_memorizing)
        plt.plot(t, data, label="value in memory")
        plt.axhline(to_memorize, c='k', lw=2, label="value to remember")
        plt.axvline(start_memorizing, c='k', ls=':', label="start gating")
        plt.legend(loc='best')

        assert abs(np.mean(data[t < 0.1])) < 0.01
        assert abs(np.mean(data[(t > 0.2) & (t <= 0.4)]) - 0.5) < 0.02
>       assert abs(np.mean(data[t > 0.4]) - 0.5) < 0.02
E       assert 0.020229506219698712 < 0.02
E        +  where 0.020229506219698712 = abs((0.47977049378030129 - 0.5))
E        +    where 0.47977049378030129 = <function mean at 0x1514c08>(array([[ 0.49574664],\n       [ 0.47269163],\n       [ 0.48927416],\n       [ 0.4...0.48226881],\n       [ 0.47013362],\n       [ 0.47461616],\n       [ 0.47927485]]))
E        +      where <function mean at 0x1514c08> = np.mean

nengo/networks/tests/test_workingmemory.py:38: AssertionError
------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------
Simulation finished in 0:00:01.                                                 
_______________________________________________________________________ test_distance _______________________________________________________________________

rng = <mtrand.RandomState object at 0x46a5e90>

    def test_distance(rng):
        a = SemanticPointer(50, rng=rng)
        b = SemanticPointer(50, rng=rng)
        assert a.distance(a) < 1e-5
>       assert a.distance(b) > 0.9
E       assert 0.77225134138145446 > 0.9
E        +  where 0.77225134138145446 = <bound method SemanticPointer.distance of <nengo.spa.pointer.SemanticPointer object at 0x46a59d0>>(<nengo.spa.pointer.SemanticPointer object at 0x46a5d10>)
E        +    where <bound method SemanticPointer.distance of <nengo.spa.pointer.SemanticPointer object at 0x46a59d0>> = <nengo.spa.pointer.SemanticPointer object at 0x46a59d0>.distance

nengo/spa/tests/test_pointer.py:149: AssertionError
____________________________________________________________ test_node_to_ensemble[nl_nodirect0] ____________________________________________________________

Simulator = <class 'nengo.simulator.Simulator'>, nl_nodirect = <class 'nengo.neurons.LIF'>, plt = <nengo.utils.testing.Mock object at 0x3e08d90>
seed = 878274874

    def test_node_to_ensemble(Simulator, nl_nodirect, plt, seed):
        N = 50

        m = nengo.Network(seed=seed)
        with m:
            m.config[nengo.Ensemble].neuron_type = nl_nodirect()
            input_node = nengo.Node(output=lambda t: [np.sin(t*3), np.cos(t*3)])
            a = nengo.Ensemble(N * 1, dimensions=1)
            b = nengo.Ensemble(N * 1, dimensions=1)
            c = nengo.Ensemble(N * 2, dimensions=2)
            d = nengo.Ensemble(N, neuron_type=nengo.Direct(), dimensions=3)

            nengo.Connection(input_node, a, function=lambda x: -x[0])
            nengo.Connection(input_node[:1], b, function=lambda x: -x)
            nengo.Connection(input_node, c, function=lambda x: -(x**2))
            nengo.Connection(input_node, d,
                             function=lambda x: [-x[0], -(x[0]**2), -(x[1]**2)])

            a_p = nengo.Probe(a, 'decoded_output', synapse=0.01)
            b_p = nengo.Probe(b, 'decoded_output', synapse=0.01)
            c_p = nengo.Probe(c, 'decoded_output', synapse=0.01)
            d_p = nengo.Probe(d, 'decoded_output', synapse=0.01)

        sim = Simulator(m)
        sim.run(2.0)
        t = sim.trange()

        plt.plot(t, sim.data[a_p])
        plt.plot(t, sim.data[b_p])
        plt.plot(t, sim.data[c_p])
        plt.plot(t, sim.data[d_p])
        plt.legend(['-sin', '-sin', '-(sin ** 2)', '-(cos ** 2)', '-sin',
                    '-(sin ** 2)', '-(cos ** 2)'], loc='best', fontsize='small')

>       assert np.allclose(sim.data[a_p][-10:], sim.data[d_p][-10:][:, 0],
                           atol=0.1, rtol=0.01)
E       assert <function allclose at 0x15180c8>(array([[ 0.44519218],\n       [ 0.36454155],\n       [ 0.31842768],\n       [ 0.3...0.35384147],\n       [ 0.40276628],\n       [ 0.33837294],\n       [ 0.34256724]]), array([ 0.35043918,  0.34762965,  0.34481699,  0.34200122,  0.33918238,\n        0.33636049,  0.33353557,  0.33070764,  0.32787674,  0.32504289]), atol=0.1, rtol=0.01)
E        +  where <function allclose at 0x15180c8> = np.allclose

nengo/tests/test_connection.py:133: AssertionError
------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------
Simulation finished in 0:00:01.                                                 
===================================================== 3 failed, 500 passed, 53 skipped in 83.49 seconds =====================================================
```

--- a/nengo/tests/conftest.py
+++ b/nengo/tests/conftest.py
@@ -1,8 +1,10 @@
 import hashlib
+import os
 
 import numpy as np
 import pytest
 
+import nengo
 import nengo.utils.numpy as npext
 from nengo.neurons import Direct, LIF, LIFRate, RectifiedLinear, Sigmoid
 from nengo.rc import rc
@@ -65,11 +67,17 @@ def plt(request):
 
 
 def function_seed(function, mod=0):
-    # take start of md5 hash of function file and name, should be pretty random
     c = function.__code__
-    to_hash = ensure_bytes(c.co_filename + c.co_name)
-    i = int(hashlib.md5(to_hash).hexdigest()[:15], 16)
-    return (test_seed + i + mod) % npext.maxint
+
+    # get function file path relative to Nengo directory root
+    nengo_path = os.path.abspath(os.path.dirname(nengo.__file__))
+    path = os.path.relpath(c.co_filename, start=nengo_path)
+
+    # take start of md5 hash of function file and name, should be unique
+    hash_list = os.path.normpath(path).split(os.path.sep) + [c.co_name]
+    hash_string = ensure_bytes('/'.join(hash_list))
+    i = int(hashlib.md5(hash_string).hexdigest()[:15], 16)
+    return (i + mod) % npext.maxint
 
 
 @pytest.fixture
@@ -78,8 +86,8 @@ def rng(request):
 
     This should be used in lieu of np.random because we control its seed.
     """
-    # add 1 to seed to be different from network seed
-    seed = function_seed(request.function, mod=1)
+    # add 1 to seed to be different from `seed` fixture
+    seed = function_seed(request.function, mod=test_seed + 1)
     return np.random.RandomState(seed)
 
 
@@ -90,7 +98,7 @@ def seed(request):
     This should be used in lieu of an integer seed so that we can ensure that
     tests are not dependent on specific seeds.
     """
-    return function_seed(request.function)
+    return function_seed(request.function, mod=test_seed)
 
 
 def pytest_generate_tests(metafunc):

--- a/nengo/tests/test_learning_rules.py
+++ b/nengo/tests/test_learning_rules.py
@@ -250,7 +250,7 @@ def test_dt_dependence(Simulator, plt, learning_rule, seed, rng):
     plt.saveas = "test_learning_rules.test_dt_dependence_%s.pdf" % (
         learning_rule.__name__)
 
-    assert np.allclose(trans_data[0], trans_data[1], atol=1e-3)
+    assert np.allclose(trans_data[0], trans_data[1], atol=2e-3)
     assert not np.all(sim.data[trans_p][0] == sim.data[trans_p][-1])
 
 

--- a/nengo/tests/test_pytest.py
+++ b/nengo/tests/test_pytest.py
@@ -1,0 +1,8 @@
+import nengo.utils.numpy as npext
+from nengo.tests.conftest import test_seed
+
+
+def test_seed_fixture(seed):
+    """The seed should be the same on all machines"""
+    i = (seed - test_seed) % npext.maxint
+    assert i == 1832276344

